tests/run-macros/i10464/Macro_1.scala

```diff
@@ -0,0 +1,16 @@
+import scala.quoted._
+
+object MatchMac {
+
+    inline def apply(inline any: Any): Unit = ${ printMacImpl('any) }
+
+    def printMacImpl(any: Expr[Any])(using Quotes): Expr[Unit] = {
+      import quotes.reflect._
+      val res = any match {
+        case '{ ($f: Person).name } => "matched!"
+        case _ => "not matched"
+      }
+      '{ println(${Expr(res)}) }
+    }
```

Comment:
The subtle semantics in quoted patterns may cause problems.

Maybe the programmer can change to Reflection for such use cases?

Or, instead of quoted patterns, add extractors to support pattern match on `Expr[T]` with clear intention and more predictable semantics.

Reply:
Yes, less common more precise matching can and must be done using reflection.

The intended semantics have not changed (https://github.com/lampepfl/dotty/issues/10464#issuecomment-733092889), it was just a bug in the implementation.

Reply:
I see, thanks for the explanation 👍